lib/administrate/page/show.rb

```diff
@@ -8,10 +8,12 @@ def initialize(dashboard, resource)
         @resource = resource
       end
 
-      attr_reader :resource
+      def resource
+        dashboard.decorate_resource(@resource)
+      end
 
       def page_title
-        dashboard.display_resource(resource)
+        dashboard.display_resource(@resource)
```

Comment:
How about having a private getter with name `#undecorated_resource`? That may help clearing confusion between `@resource` and `#resource`.